crates/relayer-components-extra/src/relay/impls/auto_relayers/parallel_bidirectional.rs

```diff
@@ -39,6 +39,9 @@ where
                 .await;
         });
 
+        // Wait for handle1 and handle2 to finish.
+        // Equivalent to Join.
+        // TODO: Confirm with Soares
         handle1.into_future().await;
```

Comment:
In any case, it might be slightly cleaner with [`tokio::join!`](https://docs.rs/tokio/latest/tokio/macro.join.html#runtime-characteristics) which will await the two handles in parallel:

```rust
tokio::join!(handle1, handle2);
```

Reply:
I think the `into_future` is used in case a runtime other than Tokio is used

Reply:
Oh I didn't know we were aiming to be runtime-agnostic. Worthy goal but a bit overkill imho. Anyway, then perhaps the spawner could expose a `join` method. Although in this context, the current solution is fine as-is.